### ISSUE TITLE
AWS SNS gateway support https endpoint

### ISSUE THREAD

**Issue by clementlecorre** [argoproj/argo-events#414]
**Is your feature request related to a problem? Please describe.**

I would like support for the https protocol for the aws sns gateway.

For the moment it is possible to configure only in http
```
error="rpc error: code = Unknown desc = failed to send subscribe request. err: InvalidParameter: Invalid parameter: Endpoint must match the specified protocol\n\tstatus code: 400"
```
AWS Support :
The protocol you want to use. Supported protocols include:
- http – delivery of JSON-encoded message via HTTP POST
- https – delivery of JSON-encoded message via HTTPS POST

**Describe the solution you'd like**
Add in the http or https event source config  : https://github.com/argoproj/argo-events/blob/master/examples/event-sources/aws-sns.yaml

**Comment by clementlecorre** [comment]
https://github.com/argoproj/argo-events/pull/415

**Comment by VaibhavPage** [comment]
available in v0.12
